### PR TITLE
chatqna ui - nginx proxy_pass

### PR DESCRIPTION
**Pull request by SeanCondon**
## Description

This adds a proxy_pass inside the nginx.conf of the `chatqna-conversation-ui`. This has the effect of avoiding CORS errors when the Javascript in the browser calls the backend service. CORS error occur when the javascript makes a call to a backend host:port not equal to where it was loaded from.

The effect is that the javascript now calls `http://host:port/v1/chatqna` which the nginx service in the pod will remap to `http://chatqna:8888/v1/chatqna` inside the cluster.

This was undertaken on the suggestion of @Ruoyu-y in [this PR 305 on GenAIInfra](https://github.com/opea-project/GenAIInfra/pull/305#issuecomment-2292560754) 

This is an additional fix for #493 and #528 

## Issues

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Load the services with docker compose according to the instructions in README.md in `docker/xeon`

### CODE REVIEW

File: ChatQnA/docker/ui/react/nginx.conf [hunk 1]
```diff
@@ -11,10 +11,17 @@ server {
...
     index index.html index.htm;
-    try_files $uri $uri/ /index.html =404;
 
     location ~* \.(gif|jpe?g|png|webp|ico|svg|css|js|mp4|woff2)$ {
       expires 1d;
     }
   }
+
+  location "/v1/chatqna" {
+      proxy_pass $CHAT_BASE_URL;
```
Comment (jaswanth8888) [inline ChatQnA/docker/ui/react/nginx.conf:21]: is the environment variable correct ?  compose.yaml is setting `APP_BACKEND_SERVICE_ENDPOINT` and `APP_DATA_PREP_SERVICE_URL` as env variables according to `https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/docker/xeon/README.md?plain=1#L403`.

Comment (SeanCondon) [reply ChatQnA/docker/ui/react/nginx.conf:21]: @jaswanth8888 thanks for pointing this out. 

In the other UI the same variable is given the [name](https://github.com/opea-project/GenAIExamples/blob/08f57fa54a3cba01ad80c76fc94e5cbed2b86c7d/ChatQnA/docker/xeon/compose.yaml#L174) `CHAT_BASE_URL`.
I think I would change this README (and the one for Gaudi and Gpu) to use the same env var name. 
* `APP_BACKEND_SERVICE_ENDPOINT` -> `CHAT_BASE_URL`

I don't see any point in having 2 names for the same thing. 
I would do the same thing for 
* `APP_DATA_PREP_SERVICE_URL` -> `UPLOAD_FILE_BASE_URL`

**Comment by jaswanth8888** [comment]
@SeanCondon  with this approach for every api that would be integrated in the UI  we have to update nginx.conf file. will this don't be a overhead when it comes to productivity suite in which all the examples would be part of single UI.

**Comment by SeanCondon** [comment]
> @SeanCondon with this approach for every api that would be integrated in the UI we have to update nginx.conf file. will this don't be a overhead when it comes to productivity suite in which all the examples would be part of single UI.

Do you have a pointer to the productivity suite? 
I think this proxy_pass for the nginx.conf will be necessary in all situations, as it is the best way for avoiding CORS restrictions securely.
The only other alternative is to have a separate load balancer container in front of the UI that will handle these redirects, but that would require the same amount of maintenance.

**Comment by jaswanth8888** [comment]
> > @SeanCondon with this approach for every api that would be integrated in the UI we have to update nginx.conf file. will this don't be a overhead when it comes to productivity suite in which all the examples would be part of single UI.
> 
> Do you have a pointer to the productivity suite? I think this proxy_pass for the nginx.conf will be necessary in all situations, as it is the best way for avoiding CORS restrictions securely. The only other alternative is to have a separate load balancer container in front of the UI that will handle these redirects, but that would require the same amount of maintenance.

@SeanCondon  https://github.com/opea-project/GenAIExamples/pull/592 please have look at this PR

**Comment by lvliang-intel** [comment]
@SeanCondon,
Maybe you can have a check about the recent changes in 'svelte' UI. We have added some endpoints for other file operations.

CHAT_BASE_URL = 'http://backend_address:8888/v1/chatqna'

UPLOAD_FILE_BASE_URL = 'http://backend_address:6007/v1/dataprep'

GET_FILE = 'http://backend_address:6007/v1/dataprep/get_file'

DELETE_FILE = 'http://backend_address:6007/v1/dataprep/delete_file'

**Comment by SeanCondon** [comment]
> @SeanCondon, Maybe you can have a check about the recent changes in 'svelte' UI. We have added some endpoints for other file operations.
> 
> CHAT_BASE_URL = 'http://backend_address:8888/v1/chatqna'
> 
> UPLOAD_FILE_BASE_URL = 'http://backend_address:6007/v1/dataprep'
> 
> GET_FILE = 'http://backend_address:6007/v1/dataprep/get_file'
> 
> DELETE_FILE = 'http://backend_address:6007/v1/dataprep/delete_file'

I have looked at this UI and I think it needs the same treatment of having a proxy.

To be honest I have tried to submit this PR and did not get any kind of positive feedback.

I do not have time to keep chasing the CI while I wait for some constructive feedback.

CC @jaswanth8888

**Comment by chensuyue** [comment]
If this is target v1.0, we need to merge by today.

**Comment by chensuyue** [comment]
Do we still need this PR? @SeanCondon

**Pull request closed without merging**